### DISCUSSION TITLE
Merge queue and optional checks

### DISCUSSION THREAD

**Discussion by SaraRavasio** [community/community#41726] [Repositories]
### Select Topic Area

Product Feedback

### Body

Hello,
I was testing out merge queue, and I think a few things are missing.

In my case, I want to 
- run lint and test coverage as required status check to be able to merge
- run integration tests only once per PR, ideally before the branch gets merged.

This would be a perfect case for merge queue, but with this set up I don't want my integration test action to be a required status check to be able to merge, because it would mean that I still need to find a way to run it before I could click on the merge button.

So my setup is:
- merge queue enabled
- cypress action `on: merge_group` but **not as a required status check**

When I tried to test it and committed a failing cypress test and tried to merge, I noticed that even if the action failed, the PR got merged anyway.

I think this really goes against the idea of merge queue, so 
- _would it be possible to block the PR from being merged if **any** of the actions defined to run in merge queue fail?_

Another issue is that if I have other required status checks, for example lint or test coverage, they would result as loading during merge queue, although they were already successful (as otherwise, I wouldn't be able to even click on the merge button).

- _could you improve merge queue to not run again the required status checks?_ 

Thank you very much and let me know if you need more details.

Sara

**Comment by bmcmanus-apex** [comment]
One way that my organization has used to work around this problem is to have a dummy workflow with the same name triggered by a pull_request event and make it a required check.  In the PR context it will run the dummy and pass allowing the merge when ready.  In the merge queue context it will run the actual tests and require they pass.  An alternative option to the dummy workflow is to have the workflow triggered by merge_group and pull_request event types.  Conditionally execute the desired steps based on the event type.  This method makes for a messier workflow but eliminates the need for a dummy workflow.

**Comment by janeklb** [comment]
Similar / same as @bmcmanus-apex, we've been solving this with dummy jobs across dedicated PR and MQ workflows.

For example, consider two workflows `pr.yaml` and `merge_queue.yaml` (that runs on `pull_request` and `merge_queue` events, respectively), and a check `ui-tests` that we only want to be required during on PRs.


```yaml
# pr.yaml
jobs:
  # ...
  ui-tests:
    # normal job definition
```


```yaml
# merge-queue.yaml
jobs:
  # ...
  ui-tests:
    runs-on: ubuntu-latest
    steps:
      - run: echo "This is a dummy job to satisfy branch protection checks"
```


That said, a native solution offered by github would be very welcome!
